### ISSUE TITLE
Some questions about the code

### ISSUE THREAD

**Issue by ZWXCV** [Michel-liu/GroupPose#10]
Dear author, thank you for your outstanding contribution. I would like to ask a few questions. Firstly, what can the checkpoint.pth file and eval.pth file be used for here? Secondly, I have tried to write visualization code for model results, but due to my lack of understanding of some parameters, many problems have arisen, making it difficult to obtain images similar to those obtained by the author in the paper, May I ask if the author can provide visual code for running out of the model? If so, it would be greatly appreciated. Looking forward to your prompt reply, thank you!

**Comment by Michel-liu** [comment]
Thanks for your interest in our work. 
The 'checkpoint.pth' file saves and can be used to resume the training status such as optimizer and model parameters. While the 'eval.pth' file preserves the evaluation results. You can check the main.py for details.

For the second question, we haven't tidied up the code for the visualization yet.
Maybe you can refer to the visualization code located at the following site.
https://github.com/leoxiaobin/deep-high-resolution-net.pytorch/blob/master/visualization/plot_coco.py

**Comment by ZWXCV** [comment]
Hello, author. I tried this code, but it requires a JSON file for predicting the results. The code you provided did not generate the corresponding file. Is there any method to generate a JSON file for predicting the results? Looking forward to your reply.

**Comment by Michel-liu** [comment]
Yes. If you directly use their code, you need to convert the output of your model into a JSON file and then run visualization. But, you can also try to integrate their visualizing part into our eval function to build a demo.

**Comment by ZWXCV** [comment]
Thank you for the author's reply. I tried to visualize in the 'evaluate' section of 'engine', but a lot of points were drawn on the graph, approximately 850: 17 x 50, which means that there are 50 instances included, which may be related to 'label'. However, in my 'result', all 'labels' are tensors of 50 ones, which do not match the ones in the targets. I found the post-processing code and found that the code used to distinguish categories is' labels=topk'_ Indexes% out_ Logits. shape [2] ',' topk_ Indexes' are all odd numbers, 'out'_ Logits. shape [2]=2 ', so all labels are 1. Logically speaking, there should be as many people as there are on the way. May I ask what may be the reason for this and how can we solve it? Looking forward to the author's recovery, thank you.

**Comment by Michel-liu** [comment]
Perhaps you should filter the output based on the predicted score rather than simply selecting the top 50.

**Comment by ZWXCV** [comment]
What is the threshold

**Comment by Michel-liu** [comment]
You can manually set it. The closer you get to 1, the fewer candidate poses there will be.

**Comment by ZWXCV** [comment]
Hello author, I am glad to receive your reply. I tried to set the score threshold to 0.4, but some of the key points of some characters in the images were not drawn. However, I set the threshold to 0.3. Although the key points on this image were drawn, others without people were also marked with points, so I am not sure how much I should set this threshold to, nor am I sure if the author has encountered similar problems, Or is the model I reproduced myself not good enough? Looking forward to the author's reply, thank you.

**Comment by Michel-liu** [comment]
Try to set a small threshold and use NMS to remove any duplication.

**Comment by ZWXCV** [comment]
In the model or where I drew the picture？

**Comment by Michel-liu** [comment]
You can modify the post-process part, using a small threshold to filter the output poses by their scores, as well as adopting NMS to remove any duplication among them.

**Comment by ZWXCV** [comment]
Ok, thank you for your reply. I'll try it first
